worker/instancepoller/worker.go

```diff
@@ -363,8 +363,19 @@ func (u *updaterWorker) pollGroupMembers(groupType pollGroupType) error {
 	}
 
 	netList, err := u.config.Environ.NetworkInterfaces(u.callContextFunc(ctx), instList)
-	if err != nil && !(errors.IsNotSupported(errors.Cause(err)) || isPartialOrNoInstancesError(err)) {
-		return errors.Trace(err)
+	if err != nil && !isPartialOrNoInstancesError(err) {
+		// NOTE(achilleasa): 2022-24-01: all existing providers (with
+		// the exception of "manual" which we don't care about in this
+		// context) implement the NetworkInterfaces method.
+		//
+		// This check is left in as a warning for folks working on new
+		// providers in the future to make sure this feature is implemented.
+		if errors.IsNotSupported(errors.Cause(err)) {
+			u.config.Logger.Warningf("substrate does not implement the required NetworkInterfaces method")
```

Comment:
I'd rather this be a straight out error return so that things break early and in your face since this is now a mandatory method. Actually, we should now add the method to the Environ interface so that it must be implemented for things to compile. The manual provider can then simply return NotSupported.